xmtp/src/conversations.rs

```diff
@@ -385,6 +393,7 @@ where
             }
         }
 
+        self.publish_outbound_payloads().await?;
```

Comment:
Unwrapping an error from publish_outbound_payloads is going to propagate all the way to the client app when something goes wrong - Is this desired? 

Is and error on publish critical or can it be logged and ignored?

Reply:
So it's not the case right now, but ideally end clients won't be calling these methods directly. I see two places where `process_outbound_messages()` will be called from the SDK:

1. On initialization after a cold start
2. On message send (`conversation.send_message()`) - likely spun off as a separate thread so that it is non-blocking to the app

In both of these places, I think an error in `publish_outbound_messages` should be propagated to `process_outbound_messages`, and then handled at the call-site inside the SDK.